Java/Graphs/KruskalsAlgorithm.java

```diff
@@ -0,0 +1,153 @@
+package Hacktoberfest2020_.Java.Graphs;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class KruskalsAlgorithm {
+    // Class to create edge
+    static class Edge{
+        int u, v, weight;
+        
+        public Edge(int u, int v, int weight){
+            this.u = u;
+            this.v = v;
+            this.weight = weight;
+        }
+        
+        @Override
+        public String toString(){
+            return this.u + " - " + this.v + " : " + this.weight;
+        }
+    }
+    
+    static class UnionFind{
+        private int parents[];
+        private int rank[];
+        
+        // Class constructor
+        public UnionFind(int x){
+            parents = new int[x];
+            rank = new int[x];
+            
+            for(int i = 0; i < x; i++){
+                parents[i] = i;
+                rank[i] = 0;
+            }
+        }
+        
+        // A utility function to find set of an element i 
+        // (uses path compression technique)
+        public int find(int node){
+            if(parents[node] != node){
+                parents[node] = find(parents[node]);
+            }
+            
+            return parents[node];
+        }
+        
+        // A function that does union of two sets of x and y 
+        // (uses union by rank) 
+        public void union(int x, int y){
+            x = find(x);
+            y = find(y);
+            
+            if(rank[x] < rank[y]){
+                parents[x] = y;
```

Comment:
You can update rank of the set too, this will limit the DSU tree height.
Just for better complexity :)